index.bs

```diff
@@ -147,35 +151,135 @@ Exactly how external changes are reflected in the data structures defined by thi
-  1. [=Wait for all=] |childPromises|, with the these success steps:
-    1. If |result| hasn't been resolved yet, [=/resolve=] |result| with `null`.
+
+  1. If |rootPath|'s [=list/size=] is greater than |childPath|'s [=list/size=],
+     [=/resolve=] |result| with null, and abort.
+
+  1. [=list/For each=] |index| of |rootPath|'s [=list/indices=]:
+    1. If |rootPath|.\[[|index|]] is not |childPath|.\[[|index|]], then
+       [=/resolve=] |result| with null, and abort.
+
+  1. Let |relativePath| be « ».
+  1. [=list/For each=] |index| of [=the range=] from |rootPath|'s [=list/size=]
+     to |rootPath|'s [=list/size=], exclusive,
+     [=list/append=] |childPath|.\[[|index|]] to |relativePath|.
+
+  1. [=/Resolve=] |result| with |relativePath|.
+
 1. Return |result|.
 
 </div>
 
+A <dfn export>file system locator</dfn> represents a potential location of a
+[=/file system entry=]. A [=/file system locator=] is either a [=file locator=]
+or a [=directory locator=].
+
+Each [=/file system locator=] has an associated <dfn export for="file system locator" id=locator-path>path</dfn> (a [=/file system path=]),
+a <dfn export for="file system locator" id=locator-kind>kind</dfn> (a {{FileSystemHandleKind}}), and
+a <dfn export for="file system locator" id=locator-root>root</dfn> (a [=file system root=]).
+
+Issue(103): Consider giving each locator a Storage Bucket.
+
+A <dfn export>file locator</dfn> is a [=/file system locator=] whose
+[=file system locator/kind=] is {{FileSystemHandleKind/"file"}}.
+A <dfn export>directory locator</dfn> is a [=/file system locator=] whose
+[=file system locator/kind=] is {{FileSystemHandleKind/"directory"}}.
+
+A <dfn export>file system root</dfn> is an opaque [=string=] whose value is
+[=implementation-defined=].
+
+<p class=example id=example-locator>For a [=/file system locator=] |locator|
+whichs [=locate an entry|locates to=] a [=file entry=] |entry| that conceptually
+exists at the path `data/drafts/example.txt` relative to the root directory of
+the [=origin private file system=],
+|locator|'s [=file system locator/kind=] has to be {{FileSystemHandleKind/"file"}},
+|locator|'s [=file system locator/path=] has to be « "`data`", "`drafts`", "`example.txt`" », and
+|locator|'s [=file system locator/root=] might include relevant identifying
+information such as the [=storage bucket=] and the disk drive.
+
+A [=/file system locator=] |a| is <dfn for="file system locator">the same locator as</dfn>
+a [=/file system locator=] |b| if
+|a|'s [=file system locator/kind=] is |b|'s [=file system locator/kind=],
+|a|'s [=file system locator/root=] is |b|'s [=file system locator/root=], and
+|a|'s [=file system locator/path=] is [=the same path as=] |b|'s [=file system locator/path=].
+
+<div algorithm>
+The <dfn export data-lt="locating an entry">locate an entry</dfn> algorithm given a
+[=/file system locator=] |locator| runs an [=implementation-defined=] series of steps adhering to
+these constraints:
+
+<ul>
```

Comment:
Thoughts on using standard markdown formatting for 
```
- lists and
  - sub-lists
```
?

This spec mostly avoids HTML if markdown suffices. Ditto with

```
create a new <code>FileSystemFileHandle</code>
create a new `FileSystemFileHandle`
```

Reply:
I think that's fine. That's really your call as editor. I wasn't sure about the syntax so I resorted to something I'm comfortable with.

Reply:
SGTM, just wanted to confirm before messing with your changes :)